PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/Invisicam.rbxmx

```diff
@@ -78,31 +98,22 @@ local function AssertTypes(param, ...)
 	assert(allowedTypes[theType], typeString .. " type expected, got: " .. theType)
 end
 
-local function CameraCast(worldPoint, ignoreList)
-	local cameraPoint = Camera.CoordinateFrame.p
-	local vector = worldPoint - cameraPoint
-	local ray = Ray.new(cameraPoint, vector.Unit * math.min(vector.Magnitude, 999))
-	return workspace:FindPartOnRayWithIgnoreList(ray, ignoreList)
-end
-
 -----------------------
 --| Local Functions |--
 -----------------------
 
 local function LimbBehavior(castPoints)
-	for _, limb in pairs(TrackedLimbs) do
-		if limb.Parent then
-			table.insert(castPoints, limb.Position)
-		end
+	for limb, _ in pairs(TrackedLimbs) do
+		castPoints[#castPoints + 1] = limb.Position
 	end
 end
 
 local function MoveBehavior(castPoints)
 	for i = 1, MOVE_CASTS do
 		local position, velocity = Torso.Position, Torso.Velocity
-		local horizontalSpeed = Vector3.new(velocity.X, 0, velocity.Z).Magnitude / 2
+		local horizontalSpeed = Vector3_new(velocity.X, 0, velocity.Z).Magnitude / 2
 		local offsetVector = (i - 1) * Torso.CFrame.lookVector * horizontalSpeed
-		table.insert(castPoints, position + offsetVector)
+		castPoints[#castPoints + 1] = position + offsetVector
```

Comment:
It's a bit faster. 

https://springrts.com/wiki/Lua_Performance#TEST_12:_Adding_Table_Items_.28table.insert_vs._.5B_.5D.29